src/argilla/client/sdk/v1/datasets/api.py

```diff
@@ -234,6 +234,28 @@ def add_records(
     """
     url = "/api/v1/datasets/{id}/records".format(id=id)
 
+    active_user_id = None
+
+    for record in records:
+        cleaned_responses = []
+        response_without_user_id = False
+        if "responses" not in record:
+            continue
+        for response in record["responses"]:
+            if response["user_id"] is None:
+                if response_without_user_id is True:
+                    warnings.warn(
+                        f"Multiple responses without `user_id` found in record {record}, so just the first one will be used while the rest will be ignored."
+                    )
+                    continue
+                else:
+                    if active_user_id is None:
+                        active_user_id = client.get("api/me").json()["id"]
+                    response["user_id"] = active_user_id
```

Comment:
maybe we can move this to the top of the method to avoid fetching the user_id for each record.